libs/h3/src/main/java/org/elasticsearch/h3/HexRing.java

```diff
@@ -581,52 +599,116 @@ public static long[] hexRing(long origin) {
         int idx = 0;
         long previous = -1;
         for (int i = 0; i < 6; i++) {
-            int[] rotations = new int[] { 0 };
-            long[] nextNeighbor = new long[] { 0 };
-            int neighborResult = h3NeighborRotations(origin, DIRECTIONS[i].digit(), rotations, nextNeighbor);
-            if (neighborResult != E_PENTAGON) {
-                // E_PENTAGON is an expected case when trying to traverse off of
+            long neighbor = h3NeighborRotations(origin, DIRECTIONS[i].digit());
+            if (neighbor != -1) {
+                // -1 is an expected case when trying to traverse off of
```

Comment:
I guess this case is actually when iterating around a pentagons neighbours we will need to skip one. Perhaps the comment could be improved?

Reply:
That is what that comment says.